align_system/algorithms/outlines_hybrid_regression_adm.py

```diff
@@ -0,0 +1,215 @@
+                    scenario_state,
+                    chosen_character
+                )
+            )
+            scenario.append(scenario_description)
+
+        # Important that the choices stay in the same order as the
+        # available actions as we'll use the selected index later to
+        # map to the corresponding action
+        choices = self.format_choices(
+            [a.unstructured for a in available_actions],
+            available_actions,
+            scenario_state
+        )
+
+        data_dict = {
+            'scenario': scenario,
+            'answer': choices
+        }
+
+        target_models = kwargs.get('models', {})
+        target_model_checkpoints = target_models["target_checkpoint"].keys()
+        model_name = target_models["model_name"]
+
+        target_kdmas = alignment_target.kdma_values
+
+        target_kdma_configs = []
+        for i, target_kdma in enumerate(target_kdmas):
+            target_kdma_name = target_kdma.kdma
+            if target_kdma_name in target_model_checkpoints:
+                log.info("Chosen KDMA: %s, and target value: %f", target_kdma_name, target_kdma.value)
+                predictions = get_kdma_predictions(
+                    model_name=model_name,
+                    data_dict=data_dict,
+                    model_path=target_models["target_checkpoint"][str(target_kdma_name)]
+                )
+                predictions = np.array([predictions])
+                dist_to_align_score = []
+                for pred in predictions:
+                    target_kdma_value = target_kdma.value
+                    dist = (target_kdma_value - pred)**2
+                    log.info("Distance to alignment target: %f", dist)
+                    dist_to_align_score.append(dist)
+
+                choice_index = np.argmin(dist_to_align_score)
+                final_choice = choices[choice_index]
+
+                target_kdma_configs.append(
+                    {
+                        "name": target_kdma_name,
+                        "score": predictions[choice_index] * 10,
+                        "action_choice": final_choice
+                    }
+                )
+
+        log.info('Predicted KDMA value from Regression model = %f', predictions[choice_index])
+        action_to_take = available_actions[choice_index]
+        action_to_take.justification = (
+            f"The direct kdma predictions from a pre-trained bert-based-uncased regression model"
+            f" for the chosen action - \"{final_choice}\" - is equal to {predictions[choice_index] * 10}."
+        )
+
+        # Set up simple diaolg to return for follow-ups
+        alignment_system_prompt = regression_error_alignment_system_prompt(target_kdma_configs)
+        prompt = action_selection_prompt(scenario_description, choices)
+        dialog = [{'role': 'system', 'content': alignment_system_prompt},
+                  {'role': 'user', 'content': prompt}]
```

Comment:
This should be the [same as in the LLM regression ADM](https://github.com/ITM-Kitware/align-system/blob/main/align_system/algorithms/outlines_regression_adm_comparative.py#L531-L534), additionally we should probably re-generate the scenario description to one that is not choice-specific:

`scenario_description = scenario_state_description_dre(scenario_state)`